services/worker-manager/src/providers/aws.js

```diff
@@ -362,8 +362,10 @@ class AwsProvider extends Provider {
     monitor.debug(`setting state to ${state}`);
     const now = new Date();
     await worker.update(this.db, worker => {
-      worker.state = state;
-      worker.lastModified = worker.state !== state ? now : null;
+      if (state !== undefined) {
```

Comment:
When would state ever be `undefined`?

Reply:
If it hasn't changed state it will still be the same value as `let state;`